ui/nuxeo-document-picker/nuxeo-document-picker.js

```diff
@@ -0,0 +1,192 @@
+            overflow: auto;
+          }
+          #topContainer {
+            display: flex;
+            justify-content: flex-end;
+          }
+          iron-icon {
+            opacity: 0.6;
+          }
+          iron-icon:hover {
+            cursor: pointer;
+            opacity: 1;
+          }
+          #mainContainer,
+          #topContainer,
+          nuxeo-results-view {
+            margin: 0;
+            padding: 0;
+          }
+          nuxeo-results-view {
+            --nuxeo-results-view-height: calc(100vh - 520px - var(--nuxeo-app-top));
+          }
+        </style>
+        <nuxeo-dialog id="dialog" modal>
+          <div id="mainContainer">
+            <div id="topContainer">
+              <iron-icon icon="nuxeo:cross" on-tap="_close"></iron-icon>
+            </div>
+            <nuxeo-results-view
+              id="resultsView"
+              provider="[[provider]]"
+              page-size="[[pageSize]]"
+              params="[[params]]"
+              quick-filters="{{quickFilters}}"
+              schemas="[[schemas]]"
+              enrichers="[[enrichers]]"
+              search-name="[[searchName]]"
+              aggregations="{{aggregations}}"
+              visible
+              show-filters
+              opened
+              search-form="[[searchForm]]"
+              on-navigate="_onNavigate"
+              on-results-changed="_onResultsChanged"
+            ></nuxeo-results-view>
+          </div>
+          <div class="buttons">
+            <!-- TODO check if we can have these stuck to the bottom (absolute positioning? check creation wizard) -->
+            <paper-button noink class="secondary" dialog-dismiss>Cancel</paper-button>
+            <paper-button noink class="primary" on-tap="_onInsert" id="insertButton">Insert</paper-button>
+          </div>
+        </nuxeo-dialog>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-picker';
+    }
+
+    static get properties() {
+      return {
+        aggregations: Object,
+        enrichers: String,
+        pageSize: Number,
+        params: Object,
+        provider: String,
+        quickFilters: Object,
+        schemas: String,
+        searchForm: Object,
+        searchName: String,
+      };
+    }
+
+    ready() {
+      super.ready();
+      if (!this._nxProvider) {
+        this._nxProvider = this.$.provider;
+      }
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      if (this.listenedResults) {
+        this.listenedResults.removeEventListener('selected-items-changed', this._updateInsertButton.bind(this));
+      }
+    }
+
+    open() {
+      if (this.$.resultsView) {
+        // XXX could make sense to consider having this as a public method on nuxeo-results-view
+        this.$.resultsView._clear();
+      }
+      this._updateInsertButton();
+      this.$.dialog.open();
+    }
+
+    _close() {
+      this.$.dialog.close();
+    }
+
+    _getSelectedItems() {
```

Comment:
But using a getter would make it public, right? Do we want that?

Reply:
the user of a getter doesn't imply public visibility, you can have `get _selectedItems() { ... }`. The thing is that imho `getXXX()` in JS is not very idiomatic since it's a language with proper support for getters and setters unlike ie Java where you need the naming convention for it...

Reply:
Going with `_selectedItems` until we can use [`#selectedItems`](https://github.com/tc39/proposal-class-fields).